federation/pkg/federation-controller/sync/controller.go

```diff
@@ -176,9 +176,15 @@ func newFederationSyncController(client federationclientset.Interface, adapter f
 			return err
 		},
 		func(client kubeclientset.Interface, obj pkgruntime.Object) error {
+			secret := obj.(*apiv1.Secret)
```

Comment:
This controller is generic - it's not intended to be just for secrets.  And a secret isn't needed here - that's what namespacedName is for.